bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/XmlFactory.java

```diff
@@ -80,8 +79,6 @@ public static BXml parse(String xmlStr) {
 
             XmlTreeBuilder treeBuilder = new XmlTreeBuilder(xmlStr);
             return treeBuilder.parse();
-        } catch (BError e) {
```

Comment:
I think this logic is correct. If you look at XMLTreeBuilder's `handleXMLStreamException` it is returning a BError value. With the removal of this, we are again creating another BError from another BError details. The other case is handing any throwable. Separating these two concerns is good.

Reply:
In the `XMLFactory`, we have two prefixes: "failed to create xml: " and "failed to parse xml: " used to start the `BError` error messages. It seems the correct thing to do is not remove the `BError` case here and also keep only a single prefix "failed to parse xml: " in the `XMLFactory`. It will be a breaking change because some error messages will be changed. Shall I do like that?

Or I can keep like this by throwing a BError at a lower level without the message prefix, and adding the prefix for both the `BError` and `Throwable` here when creating a new BError.

Reply:
Went with the first approach as said in the review comment. Fixed with 30ffe3cdcee90ee72db3e1ade17408e58ab31ec4.